File: frontend/src/types/internalConsult.ts
export type ConsultCaseType = "surgical" | "gyne" | "nongyne";
export type ConsultStatus = "pending" | "responded" | "closed";

export interface UserMini {
  id: number;
  full_name?: string;
  report_name?: string;
}

export interface InternalConsult {
  id: number;
  case_type: ConsultCaseType;
  report_id: number;
  requester_id: number;
  consultant_id: number;
  reason: string;
  opinion?: string | null;
  accession_no_snapshot?: string | null;
  status: ConsultStatus;
  promoted_to_signer: boolean;
  created_at: string;
  responded_at?: string | null;
  closed_at?: string | null;
  requester?: UserMini;
  consultant?: UserMini;
}

export interface InternalConsultCreate {
  case_type: ConsultCaseType;
  report_id: number;
  consultant_id: number;
  reason: string;
}

export interface InternalConsultRespondRequest {
  opinion: string;
}

export interface InternalConsultPromoteRequest {
  role?: string;
  consult_note?: string;
}

export interface InternalConsultListResponse {
  items: InternalConsult[];
  total: number;
}

File: backend/app/schemas/internal_consult.py
from pydantic import BaseModel, ConfigDict
from datetime import datetime
from typing import Optional, List


class UserMiniResponse(BaseModel):
    id: int
    full_name: Optional[str] = None
    report_name: Optional[str] = None

    model_config = ConfigDict(from_attributes=True)


class InternalConsultCreate(BaseModel):
    case_type: str  # "surgical" | "gyne" | "nongyne"
    report_id: int
    consultant_id: int
    reason: str


class InternalConsultRespondRequest(BaseModel):
    opinion: str


class InternalConsultPromoteRequest(BaseModel):
    role: str = "co-signer"
    consult_note: Optional[str] = None


class InternalConsultResponse(BaseModel):
    id: int
    case_type: str
    report_id: int
    requester_id: int
    consultant_id: int
    reason: str
    opinion: Optional[str] = None
    accession_no_snapshot: Optional[str] = None
    status: str
    promoted_to_signer: bool
    created_at: datetime
    responded_at: Optional[datetime] = None
    closed_at: Optional[datetime] = None
    requester: Optional[UserMiniResponse] = None
    consultant: Optional[UserMiniResponse] = None

    model_config = ConfigDict(from_attributes=True)


class InternalConsultListResponse(BaseModel):
    items: List[InternalConsultResponse]
    total: int
